cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -368,7 +416,11 @@ private Builder(Schema schema) {
      *                                   value is given
      */
     public Builder set(String fieldName, @Nullable Object value) {
-      validateAndGetField(fieldName, value);
```

Comment:
There are existing methods for validation that happens in the code, 
1. validateAndGetField - validates at Field Object 
2. validateAndGetLogicalTypeSchema - validates schema object for logical type constraints

Fixed type needs 2 levels of validation for set. 
1. Schema check which can be reused from validateAndGetLogicalTypeSchema. 
2. Value check, length should not exceed the fixed size(which is implemented as part of setFixedLogicalTypeSchema).

Should we restructure the Generic set function a bit to accommodate all these requirements? 

public Builder set(String fieldName, @Nullable Object value) {
Schema.Field field = validateAndGetField(fieldName, value); // existing line 
validateSchema(field.getSchema()); // new function but it could invoke validateAndGetLogicalTypeSchema for logical types. I assume there are no schema checks for other types today. 
validateValue(field.getSchema(), value); // new function which would invoke setFixedLogicalTypeSchema(without fields.put line) for logical types and i assume there no value checks for other types today. 
}

Both new functions can be extended for type specific checks in future if there is any. 

@fernst Please share your thoughts here.